src/lsstseries/ensemble.py

```diff
@@ -475,7 +580,14 @@ def from_parquet(
 
             # Generate an object table from the source table, then merge
             generated = self._generate_object_table()
-            self._object = file_table.merge(generated, how="right", on=[self._id_col])
+            self._object = file_table.merge(
```

Comment:
The id column may not be the right column to join on, when merging two tables, and hipscat tables will likely use the `_hipscat_index` as the pandas dataframe index for performance reasons.

Reply:
the `read_parquet` call reindexes based on whichever column was assigned to `self._id_col`, and in this instance the join is between the input object files and an object table generated from the source table. So I *think* the id column makes sense here. You're right in that because it isn't the index it will be slower, but I'm not quite sure what we could do to circumvent that.